### PR TITLE
[Glimmer] Unskip component registration tests

### PR DESCRIPTION
**Pull request by nickiaconis**
Title says it all.

### CODE REVIEW

File: packages/ember-views/tests/utils/helpers.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import isEnabled from 'ember-metal/features';
+import require from 'require';
+
+export let OutletView;
+
+if (isEnabled('ember-glimmer')) {
+  OutletView = require('ember-glimmer/views/outlet').default;
+} else {
+  OutletView = require('ember-htmlbars/views/outlet').OutletView;
```
Comment (krisselden) [inline packages/ember-views/tests/utils/helpers.js]: ember-templates already does this branch

Comment (krisselden) [inline packages/ember-views/tests/utils/helpers.js]: nevermind, I thought it did, but I was wrong. I'm trying to get away from ember-views requiring ember-glimmer or ember-htmlbars, I'm trying to have it just be where stuff that is shared not branched is.

Comment (krisselden) [inline packages/ember-views/tests/utils/helpers.js]: if this is used from a test in packages/ember/tests only can you move it there?

Comment (nickiaconis) [inline packages/ember-views/tests/utils/helpers.js]: @krisselden Makes sense. I've moved the file into the ember package.

**Comment by chadhietala** [comment]
LGTM

File: packages/ember-glimmer/lib/utils/lookup-component.js [hunk 2]
```diff
@@ -7,14 +8,40 @@ function lookupComponentPair(componentLookup, owner, name, options) {
...
 
-  if (!layout && component) {
-    let layoutProp = component.proto().layout;
-    if (layoutProp) {
-      let templateFullName = 'template:components/' + name;
-      owner.register(templateFullName, layoutProp);
+  if (component) {
+    let templateFullName = 'template:components/' + name;
+    let {
+      templateForName,
```
Comment (krisselden) [inline packages/ember-glimmer/lib/utils/lookup-component.js]: why do we mixin this into component and pull it off? can we just drop it

File: packages/ember-glimmer/lib/utils/lookup-component.js [hunk 3]
```diff
@@ -7,14 +8,40 @@ function lookupComponentPair(componentLookup, owner, name, options) {
...
+      owner.register(templateFullName, defaultLayoutProp);
       layout = owner.lookup(templateFullName, options);
-    } else {
-      layout = owner.lookup(DEFAULT_LAYOUT);
+    } else if (!layout) {
+      if (layoutProp) {
+        owner.register(templateFullName, layoutProp);
+        layout = owner.lookup(templateFullName, options);
+      } else if (layoutNameProp) {
+        layout = templateForName(layoutNameProp, null, owner);
```
Comment (krisselden) [inline packages/ember-glimmer/lib/utils/lookup-component.js]: this is weird, lets just move the templateForName logic to the mixin that is actually calling it.

File: packages/ember-glimmer/lib/utils/lookup-component.js (hunk 2 above)
Comment (nickiaconis) [inline packages/ember-glimmer/lib/utils/lookup-component.js]: @krisselden Htmlbars expects it to be on the view/component, but component init runs too late for the same logic to apply in glimmer. I was trying to keep things DRY by pulling templateForName off the class, but I can WET-ify that function if it's preferred.

Comment (krisselden) [inline packages/ember-glimmer/lib/utils/lookup-component.js]: I think we should emit a different opcode or the dynamic opcode when a layout is late bound and assert on the layout property like the tag expression is asserted.

Comment (krisselden) [inline packages/ember-glimmer/lib/utils/lookup-component.js]: maybe just use the partial opcode for the layout

**Comment by chadhietala** [comment]
Superseded by #13588.

**Pull request closed without merging**